components/tidb_query/src/batch/runner.rs

```diff
@@ -404,7 +419,14 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     .collect_exec_stats(&mut self.exec_stats);
 
                 let mut sel_resp = SelectResponse::default();
-                sel_resp.set_chunks(chunks.into());
+                match self.config.encode_type {
```

Comment:
Why not use the same field? It is byte array anyway and it should be suitable for both formats.

Reply:
Now use the different fields can also check the EncodeType. (One of the fields is empty.) If the TiKV is not support the feature, It can also use the right DecodeType in TiDB. 
In fact, it can use the same field, but need a flag to get the EncodeType. In future, i will change the protobuf(add a flag and remove the field `row_batch_data`), and change the code to use the same field.

Reply:
Is `row_batch_data` a newly added field? Then maybe we can just add a `response_format` instead of `row_batch_data`?

Reply:
Yes.